Assets/Scripts/Hierarchical/HierarchicalBase.cs

```diff
@@ -59,27 +63,38 @@ public void RemovePursuer(IHierarchical pursuer) {
     _pursuers.Remove(pursuer);
   }
 
-  protected virtual Vector3 GetPosition() {
-    return GetMean(s => s.Position);
-  }
-
-  protected virtual Vector3 GetVelocity() {
-    return GetMean(s => s.Velocity);
-  }
-
-  protected virtual Vector3 GetAcceleration() {
-    return GetMean(s => s.Acceleration);
+  public bool IsEscapingPursuers() {
+    return Pursuers.All(pursuer => {
+      // A hierarchical object is considered escaping a pursuer if the closing velocity is
+      // non-positive or if the hierarchical object will reach its target before the pursuer reaches
+      // the hierarchical object.
+      Vector3 relativePosition = pursuer.Position - Position;
+      Vector3 relativeVelocity = pursuer.Velocity - Velocity;
+      float rangeRate = Vector3.Dot(relativeVelocity, relativePosition.normalized);
+      float closingVelocity = -rangeRate;
+      if (closingVelocity <= 0) {
+        return true;
+      }
+      float pursuerDistance = relativePosition.magnitude;
+      float targetDistance = (Target.Position - Position).magnitude;
+      float timeToTarget = targetDistance / Speed;
+      float pursuerTimeToIntercept = pursuerDistance / pursuer.Speed;
```

Comment:
Potential div by zero if speed is zero (could happen momentarily if a quadcopter perfectly reverses direction)
Suggest
```cs
if (Speed <= 0 || pursuer.Speed <= 0) {
  return false; // Cannot escape if not moving
}
```